main.go

```diff
@@ -86,6 +87,10 @@ func main() {
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
+	if err := util.InspectCluster(); err != nil {
+		setupLog.Info("unable to inspect cluster")
+	}
```

Comment:
Can we safely assume that the availability of the console API will never change during the lifecycle of the cluster?

For example, can the console API be provisioned to the cluster, after it has been installed without the API being available? If this case, the Operator would need to be restarted to pick up the API's availability.

Reply:
Not sure what to do here.  We could check for the existence of the console API every time we reconciled, but that would still require a reconcile to be triggered before the operator would pick up the API's availability.  It would also introduce a bit of a performance slowdown for everyone (but maybe that would be negligible).  I don't know of any way to get kubernetes to alert us when new API is added to the system.

@jannfis do you have any suggestions?

Reply:
I think it's fine to keep as is, knowing about this possible caveat. If ConsoleLink API is something that can added/removed at will (and not only at installation time), I think it's a fair workaround to have users simply restart the Operator.